app/components/Views/NetworkConnect/NetworkConnectMultiSelector/NetworkConnectMultiSelector.tsx

```diff
@@ -145,60 +66,34 @@ const NetworkConnectMultiSelector = ({
       isSelected: false,
       //@ts-expect-error - The utils/network file is still JS and this function expects a networkType, and should be optional
       imageSource: getNetworkImageSource({
-        chainId: network?.chainId,
+        chainId: network.chainId,
       }),
+      chainId: network.chainId,
     }),
   );
 
-  const onSelectNetwork = useCallback(
-    (clickedChainId: string) => {
-      const selectedAddressIndex = selectedChainIds.indexOf(clickedChainId);
-      // Reconstruct selected network ids.
-      const newNetworkList = networks.reduce((acc, { id }) => {
-        if (clickedChainId === id) {
-          selectedAddressIndex === -1 && acc.push(id);
-        } else if (selectedChainIds.includes(id)) {
-          acc.push(id);
-        }
-        return acc;
-      }, [] as string[]);
-      setSelectedChainIds(newNetworkList);
-    },
-    [networks, selectedChainIds],
-  );
-
-  const onRevokeAllHandler = useCallback(async () => {
-    await Engine.context.PermissionController.revokeAllPermissions(hostname);
-    navigate('PermissionsManager');
-  }, [hostname, navigate]);
-
-  const toggleRevokeAllNetworkPermissionsModal = useCallback(() => {
-    navigate(Routes.MODAL.ROOT_MODAL_FLOW, {
-      screen: Routes.SHEET.REVOKE_ALL_ACCOUNT_PERMISSIONS,
-      params: {
-        hostInfo: {
-          metadata: {
-            origin: urlWithProtocol && new URL(urlWithProtocol).hostname,
-          },
-        },
-        onRevokeAll: !isRenderedAsBottomSheet && onRevokeAllHandler,
-      },
-    });
-  }, [navigate, urlWithProtocol, isRenderedAsBottomSheet, onRevokeAllHandler]);
+  const onSelectNetwork = useCallback((chainId: string) => {
+      if (selectedChainIds.includes(chainId)) {
+        setSelectedChainIds(
+          selectedChainIds.filter((_chainId) => _chainId !== chainId),
+        );
+      } else {
+        setSelectedChainIds([...selectedChainIds, chainId]);
+      }
+  }, [selectedChainIds, setSelectedChainIds]);
 
   const areAllNetworksSelected = networks
-    .map(({ id }) => id)
-    .every((id) => selectedChainIds?.includes(id));
-  const areAnyNetworksSelected = selectedChainIds?.length !== 0;
-  const areNoNetworksSelected = selectedChainIds?.length === 0;
+    .every(({ chainId }) => selectedChainIds.includes(chainId));
+  const areAnyNetworksSelected = selectedChainIds.length !== 0;
+  const areNoNetworksSelected = selectedChainIds.length === 0;
```

Comment:
[nit]: slightly more easily readable suggestion. Leave to your discretion since this is more subjective than objective

```suggestion
const areAnyNetworksSelected = selectedChainIds.length > 0;
const areNoNetworksSelected = !areAnyNetworksSelected;
```

Reply:
i didn't write these lol

Reply:
Ignore this then, mb